source/dem/find_contact_detection_frequency.cc

```diff
@@ -0,0 +1,69 @@
+#include <dem/find_contact_detection_frequency.h>
+
+using namespace dealii;
+
+template <int dim>
+bool
+find_contact_detection_frequency(
+  Particles::ParticleHandler<dim> &particle_handler,
+  const double &                   neighborhood_threshold,
+  const double &                   minimum_cell_size,
+  const double &                   dt,
+  const double &                   maximum_particle_diameter,
+  const double &                   dynamic_contact_search_factor)
+{
+  bool   update_step      = false;
+  double max_displacement = 0;
+
+  // Looping through all the particles:
+  for (auto &particle : particle_handler)
+    {
+      auto &particle_properties = particle.get_properties();
+
+      for (unsigned int d = 0; d < dim; ++d)
+        {
+          particle_properties[DEM::PropertiesIndex::displacement_x + d] +=
+            particle_properties[DEM::PropertiesIndex::v_x + d] * dt;
+
+          max_displacement = std::max(
+            max_displacement,
+            std::abs(
+              particle_properties[DEM::PropertiesIndex::displacement_x + d]));
```

Comment:
This is weird. The real displacement is the norm of the displacement of the particles. Now you are looking the max displacement in every direction, it should be the norm of the displacement. The max should be outside of the loop. The formula used right now is wrong.